src/huggingface_hub/hf_api.py

```diff
@@ -1845,13 +1854,20 @@ def upload_file(
 
         path = f"{self.endpoint}/api/{repo_id}/upload/{revision}/{path_in_repo}"
 
-        headers = {"authorization": f"Bearer {token}"} if token is not None else None
+        headers = {"authorization": f"Bearer {token}"} if token is not None else {}
```

Comment:
Here I change the default value of `headers` from `None` to an empty `dict` to allow me to inject the commit fields when `create_pr` is `True`